mhkit/tests/Utils_TestGenUtils.m

```diff
@@ -71,6 +73,142 @@ function test_magnitude_phase(testCase)
+            %2. Check Group Attributes
+            %fprintf("2. Check Group Attributes: ");
+            finfo = ncinfo(strcat('example_ncfiles/',fnm));
+            val1 = res.groups.METADATA.groups.ALGORITHM_SETTINGS.groups.SLANT_COLUMN_RETRIEVAL.Attributes;
+            val2 = finfo.Groups(2).Groups(1).Groups(1).Attributes;
+            assertEqual(testCase,val1,val2);
+            %3. Check Variables
+            %fprintf("3. Check Variable \n");
+            % file with groups
+            %fprintf("Check file with groups\n");
+            % '/PRODUCT/SUPPORT_DATA/DETAILED_RESULTS'
+            ginfo = finfo.Groups(1).Groups(1).Groups(2);
+            vnms = {ginfo.Variables.Name};
+            sz = size(ginfo.Variables);
+            % 3.1 check Dims
+            %fprintf("3.1 Check Variable Dims: ");
+            idx = randi([1,sz(2)],1); 
+            vname = check_name(vnms{idx});
+            val1 = res.groups.PRODUCT.groups.(['SUPPORT_' ...
+                'DATA']).groups.DETAILED_RESULTS.Variables.(vname).Dims;
+            val2 = {ginfo.Variables(idx).Dimensions.Name};
+            val3 = size(res.groups.PRODUCT.groups.(['SUPPORT_' ...
+                'DATA']).groups.DETAILED_RESULTS.Variables.(vname).Data);
+            val4 = size(ncread(strcat('example_ncfiles/',fnm),...
+                strcat('PRODUCT/SUPPORT_DATA/DETAILED_RESULTS/',vnms{idx})));
+            assertEqual(testCase,val1,val2);
+            assertEqual(testCase,val3,val4);
+            % 3.2 check Data
+            %fprintf("3.2 Check Variable Data: ");
+            idx = randi([1,sz(2)],1); 
+            vname = check_name(vnms{idx});
+            val1 = res.groups.PRODUCT.groups.(['SUPPORT_' ...
+                'DATA']).groups.DETAILED_RESULTS.Variables.(vname).Data;
+            val2 = ncread(strcat('example_ncfiles/',fnm),...
+                strcat('PRODUCT/SUPPORT_DATA/DETAILED_RESULTS/',vnms{idx}));
+            
+            testCase.verifyTrue(isequaln(val1,val2),vname);
+            % 3.3 check Attributes
+            %fprintf("3.3 Check Variable Attributes: \n");
+            idx = randi([1,sz(2)],1); 
+            vname = check_name(vnms{idx});
+            in_names = {ginfo.Variables(idx).Attributes.Name};
+            in_vals = {ginfo.Variables(idx).Attributes.Value};
+            xtemp = res.groups.PRODUCT.groups.(['SUPPORT_' ...
+                'DATA']).groups.DETAILED_RESULTS.Variables;
+            for iattr = 1:numel(in_names)
+                %fprintf(" - Attr %s: ",in_names{iattr});
+                if strcmp(in_names{iattr},'_FillValue')
+                    out_val = xtemp.(vname).FillValue;
+                else
+                    out_val = xtemp.(vname).Attrs.(in_names{iattr});
+                end
+                testCase.verifyTrue(isequaln(in_vals{iattr},out_val));
+            end
+
+            % file without groups
+            %fprintf("Check file without groups\n");
+            for ifnm = 3:numel(fnms)
```

Comment:
If only intending to test the single file called out below, you can remove this loop